utils/build_swift/driver_arguments.py

```diff
@@ -0,0 +1,1295 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import argparse
+import multiprocessing
+import os.path
+
+import android.adb.commands
+
+# TODO: Replace with implementation in build_swift package
+from swift_build_support.swift_build_support import host
+from swift_build_support.swift_build_support.targets import \
+    StdlibDeploymentTarget
+
+from . import defaults
+from .argparser_builder import ArgParserBuilder
+
+
+__all__ = [
+    'create_argument_parser',
+]
+
+
+def create_argument_parser():
+    """Return configured argument parser for main driver."""
+
+    # NOTE: USAGE, DESCRIPTION and EPILOG are defined at the bottom of the file
+    builder = ArgParserBuilder(
+        usage=USAGE,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=DESCRIPTION,
+        epilog=EPILOG)
+
+    # Prepare DSL functions
+    in_group = builder.in_group
+    option = builder.add_option
+    set_defaults = builder.set_defaults
+    mutually_exclusive_group = builder.mutually_exclusive_group
+
+    append = builder.append_action
+    disable = builder.disable_action
+    enable = builder.enable_action
+    set_ = builder.set_action
+    unsupported = builder.unuspported_action
+
+    def realpath(path):
+        path = os.path.expanduser(path)
+        path = os.path.abspath(path)
+        path = os.path.realpath(path)
+        return path
+
+    # -------------------------------------------------------------------------
+    # Top-level options
+
+    option(['-n', '--dry-run'], enable, default=False, nargs=0,
+           help='print the commands that would be executed, but do not '
+                'execute them')
+
+    option(['-c', '--clean'], enable, default=False, nargs=0,
+           help='clean the build directory')
+
+    # TODO: rename to '-v' and '--verbose'
+    option('--verbose-build', enable('verbose'), default=False, nargs=0,
+           help='print the commands executed during the build of each project')
+
+    option('--reconfigure', enable, default=False, nargs=0,
+           help='force a CMake configuration run even if CMakeCache.txt '
+                'already exists')
+
+    option('--show-sdks', enable, default=False, nargs=0,
+           help='print the installed Xcode and SDK versions')
+
+    option('--no-legacy-impl', disable('lecacy_impl'), default=True, nargs=0,
+           help='avoid legacy build-script-impl implementation')
+
+    option('--export-compile-commands', enable, default=False, nargs=0,
```

Comment:
The default for this should be true. It is not expensive and can be useful for tooling purposes.

Reply:
For the purpose of this PR I am hesitant to change any default values since I'm already replacing the whole argument parser. I'd rather hold off on reworking the defaults to be more sensible until we have this merged.

Reply:
I thought the default was to have this enabled.

Reply:
I can double check as I make a cleanup pass over this again to make sure all the defaults are preserved (in combination with the new unit-tests).